exchange/exchange.go

```diff
@@ -606,14 +607,20 @@ func updateHbPbCatDur(bid *entities.PbsOrtbBid, dealTier openrtb_ext.DealTier, b
 	}
 }
 
-func (e *exchange) makeAuctionContext(ctx context.Context, needsCache bool) (auctionCtx context.Context, cancel context.CancelFunc) {
+func (e *exchange) makeAuctionContext(ctx context.Context, needsCache bool, tmaxAdjustments *config.TmaxAdjustments) (auctionCtx context.Context, cancel context.CancelFunc) {
 	auctionCtx = ctx
 	cancel = func() {}
 	if needsCache {
 		if deadline, ok := ctx.Deadline(); ok {
 			auctionCtx, cancel = context.WithDeadline(ctx, deadline.Add(-e.cacheTime))
 		}
 	}
+	if tmaxAdjustments != nil && tmaxAdjustments.Enabled {
```

Comment:
No need to subtract `UpstreamResponseDuration` from `deadline` if `UpstreamResponseDuration` is zero. Should add `UpstreamResponseDuration > 0` check

```
if tmaxAdjustments != nil && tmaxAdjustments.Enabled && tmaxAdjustments.UpstreamResponseDuration > 0{
```

Reply:
https://github.com/prebid/prebid-server/pull/2750/commits/3c5167ea79f099afb0eeaf1f5398f4b444c4135c